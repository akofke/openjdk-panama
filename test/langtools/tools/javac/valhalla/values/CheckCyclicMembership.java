/*
 * @test /nodynamiccopyright/
 * @summary Value types may not declare fields of its own type either directly or indirectly.
 *
 * @compile/fail/ref=CheckCyclicMembership.out -XDenableValueTypes -XDrawDiagnostics CheckCyclicMembership.java
 */

final __ByValue class CheckCyclicMembership {
    class InnerRef {
        CheckCyclicMembership ccm;
    }
    __ByValue final class InnerValue {
        final CheckCyclicMembership ccm = __MakeDefault CheckCyclicMembership(); // Error.
    }
    final CheckCyclicMembership ccm = __MakeDefault CheckCyclicMembership(); // Error.
    final int i = 10;
    final String s = "blah";
    final InnerRef ir = new InnerRef(); // OK.
    final InnerValue iv = __MakeDefault InnerValue(); // Error. Some Order dependancy hides this. FIXME.
}
