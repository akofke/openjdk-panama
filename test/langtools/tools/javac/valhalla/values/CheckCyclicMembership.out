CheckCyclicMembership.java:15:33: compiler.err.cyclic.value.type.membership: CheckCyclicMembership
CheckCyclicMembership.java:13:37: compiler.err.cyclic.value.type.membership: CheckCyclicMembership.InnerValue
2 errors
